src/components/DownloadModal/index.tsx

```diff
@@ -0,0 +1,414 @@
+import React, { Component } from 'react';
+// tslint:disable-next-line
+import Modal from 'react-modal';
+import { Colors, Styles, blueGradientBgImg } from '../../constants';
+import { TaskButton } from './TaskButton';
+import _ from 'lodash';
+// tslint:disable: no-unsafe-any no-any;
+/**
+ * Properties for DownloadModal
+ * @export
+ * @interface DownloadModalProps
+ */
+export interface DownloadModalProps {
+  taskModels?: string[];
+  isOpen: boolean;
+}
+/**
+ * State for DownloadModal
+ * @export
+ * @interface DownloadModalProps
+ */
+export interface DownloadModalState {
+  showModal: boolean;
+  showHide: string;
+  showMultiSelect: string;
+  taskModelButtons: JSX.Element[];
+  selectedList: JSX.Element[];
+  confirmArr: string[];
+}
+
+const customStyles = {
+  content: {
+    top: '50%',
+    left: '50%',
+    right: 'auto',
+    bottom: 'auto',
+    marginRight: '-50%',
+    transform: 'translate(-50%, -50%)'
+  }
+};
+
+if (process.env.NODE_ENV !== 'test') Modal.setAppElement('body');
+
+/**
+ * Renders a Modal window to download a document's targets
+ * @export
+ * @class {DownloadModal}
+ * @param {DownloadModalProps} item
+ */
+export class DownloadModal extends Component<DownloadModalProps, DownloadModalState> {
+  constructor(props: DownloadModalProps) {
+    super(props);
+    this.state = {
+      showModal: props.isOpen,
+      showHide: 'hidden',
+      showMultiSelect: '',
+      taskModelButtons: this.generateButtons(props),
+      selectedList: [],
+      confirmArr: []
+    };
+    this.closeModal = this.closeModal.bind(this);
+    this.handleContinue = this.handleContinue.bind(this);
+    this.handleBackButton = this.handleBackButton.bind(this);
+    this.handleConfirm = this.handleConfirm.bind(this);
+  }
+  /**
+   * Handles the user clicking the Confirm Button
+   *
+   * To Do: Link result to PDF generation/download
+   */
+
+  handleConfirm() {
+    const selectedArr: string[] = [];
+    this.state.taskModelButtons.forEach(task => {
+      if (task.props.toggled === true) {
+        selectedArr.push(task.props.taskName);
+      }
+    });
+    this.setState({ confirmArr: selectedArr });
+    this.closeModal();
+  }
+
+  closeModal() {
+    this.setState({ showModal: false });
+  }
+
+  /**
+   * Handles the user clicking the Continue Button
+   *
+   * Toggles hidden selections page and makes an unordered list of selected buttons
+   */
+
+  handleContinue() {
+    const taskButtons = _.cloneDeep(this.state.taskModelButtons);
```

Comment:
If we change the state to represent the data we need rather than the whole element we won't need to do this. Generally we shouldn't need to make deep copies of state.